### PR TITLE
"Imviz Line Profiles (XY)" renamed to "Image Profiles (XY)"

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request renames "Imviz Line Profiles (XY)" to ~"Line Profiles (XY)"~ "Image Profiles (XY)" along with deprecation for API access (although the plugin API itself is still not exposed).

~If we want to name this to something more general (especially if we would ever want to re-use this for 2d spectra and want to avoid confusion with spectral lines) now would be the time to do so.~

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/helpers.py [hunk 1]
```diff
@@ -137,13 +137,20 @@ def plugins(self):
...
+            # renamed in 3.9
             plugins['Links Control'] = plugins['Orientation']._obj.user_api
             plugins['Links Control']._deprecation_msg = 'in the future, the formerly named \"Links Control\" plugin will only be available by its new name: \"Orientation\".'  # noqa
         if 'Canvas Rotation' in plugins:
+            # removed in 3.9
             plugins['Canvas Rotation']._deprecation_msg = 'this functionality will be removed in favor of the implementation for rotation in the \"Orientation\" plugin.'  # noqa
         if 'Export' in plugins:
+            # renamed in 3.9
             plugins['Export Plot'] = plugins['Export']._obj.user_api
             plugins['Export Plot']._deprecation_msg = 'in the future, the formerly named \"Export Plot\" plugin will only be available by its new name: \"Export\".'  # noqa
```
Comment (kecnry) [inline jdaviz/core/helpers.py:149]: I think these have now been more than 2 releases, so maybe we can remove for 4.0?

**Comment by pllim** [comment]
Any downstream notebooks need updating to go with this?

File: CHANGES.rst [hunk 2]
```diff
@@ -32,6 +32,8 @@ Imviz
 
 - Added a table with catalog search results. [#2915, #3101]
 
+- "Line Profiles XY" plugin is renamed to "Profiles XY". [#3121]
```
[the remark this answers is not in the document]
Comment (kecnry) [reply CHANGES.rst:35]: woops, yes, thanks!

**Comment by kecnry** [comment]
> Any downstream notebooks need updating to go with this?

Not that I know, there is currently no API access besides showing/opening the plugin, so I doubt any notebook makes use of this.  If it does, they'll start seeing the "deprecation" message and hopefully know how to update before we remove that.

File: jdaviz/core/helpers.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/core/helpers.py:149]: (I don't think that belongs in scope here, so I made a note in the 4.0 release ticket)

**Review by pllim — APPROVED** [review]
Looks uncontroversial enough. Thanks!

**Comment by kecnry** [comment]
(let's not merge until we have a chance to discuss generalizing the name - just canceled auto-merge)

**Comment by pllim** [comment]
Oh OK... I thought the ticket was pretty clear. Sorry if I misunderstood.

**Comment by kecnry** [comment]
@pllim - I did find one mention in a concept notebook and updated that here along with changing to "Image Profiles (XY)" as discussed offline.  Are you ok with marking this as trivial and merging from your previous approval?

File: jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 __all__ = ['LineProfileXY']
 
 
-@tray_registry('imviz-line-profile-xy', label="Imviz Line Profiles (XY)")
+@tray_registry('imviz-line-profile-xy', label="Line Profiles (XY)")
```
Comment (pllim) [inline jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py:14]: Typo?
```suggestion
@tray_registry('imviz-line-profile-xy', label="Image Profiles (XY)")
```

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py]: ah, I had that locally but somehow missed staging it 🤦 .  I'll take another look at the final diff to make sure I didn't miss anything else, they're so easy to miss.

Comment (pllim) [reply jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py]: Let me know when you done. Thanks!

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py]: everything looks right to me now, thanks!  Let me know if you see anything else

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**